### PR TITLE
Changes D&Dv2

### PR DESCRIPTION
**Pull request by lcamacho**
## Description
These changes include:
- Remove max-width
- Fix title css classes
- Hides text from draggable until gets processed after drop
- Change containers and css classes to use those from openedx 
- Feature: add option to show zone border when dragging an item 
borders-dragging

- Feature: add item option to remove padding from images dropped on a zone
no-padding

- Improve feedback styles, now only icons shows color related with feedback and text color remains the same also each feedback line doesn't show borders.
Captura de Pantalla 2022-03-31 a la(s) 8 45 50 p  m

## Testing instructions
To test the way it hides text until drops gets processed:
1. With the standard demo of drag and drop v2, grab the text item that belongs to top
2. Drop it into the top zone and see how the text gets replaced by the spinner icon but doesn't change item size, once it's processed the text should appear back

These instructions are for testing the new added features.
Remove padding:
1. Add an image to the assets, let's say for this example an image of 60x60px
3. Add a drag and drop v2 xblock 
4. Edit configuration, add a zone of 60x60px or the image size
5. Add the image as a new item and check "Remove padding when dropped on a zone (applies only to images)"
6. Finally, drag that image into that zone, and you should see how that fit well without any padding 

Add border to drag zone:
1. Edit configuration, move onto zones configuration and check "Display zone borders when dragging an item"
2. Drag some element into the zones, and you will see a border added to each zone when dragging over them. Note that when ""Display zone borders on the image" and "Display zone borders when dragging an item" the first takes precedence and will only display the regular borders on the zone.

Supporting information
None

Deadline
None

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @lcamacho! I've created [OSPR-6468](https://openedx.atlassian.net/browse/OSPR-6468) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by lcamacho** [comment]
@natabene ready for review

**Comment by natabene** [comment]
@lcamacho Thanks for letting me know. @bradenmacdonald Can someone from your team review this, as you know it best?
 CC @sarina Are you ok with this?

**Comment by Agrendalath** [comment]
@lcamacho, I'll let @0x29a review this first. In the meantime, please bump the version in `setup.py` and update the changelog.

File: drag_and_drop_v2/public/css/drag_and_drop.css
Comment (farhaanbukhsh) [inline drag_and_drop_v2/public/css/drag_and_drop.css:866]: New line missing

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (farhaanbukhsh) [inline drag_and_drop_v2/public/js/drag_and_drop.js:118]:
```suggestion
            style.maxWidth = (possiblemaxWidth < 100) ? possiblemaxWidth + '%'  :  '100%';
```

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (farhaanbukhsh) [inline drag_and_drop_v2/public/js/drag_and_drop.js:1528]: is the naming of the variable intentional? Else it creates a confusion with jQuery syntax

**Review by farhaanbukhsh — CHANGES_REQUESTED** [review]
I have left a few comments on the PR and on the Jira ticket, please have a look @lcamacho , good work so far.

**Comment by sarina** [comment]
@natabene OpenCraft knows more about this XBlock than my team does so I defer to them.

@Agrendalath does your team have core committers on this repo? If not we should go through the nomination process for a few of your team.

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (lcamacho) [reply drag_and_drop_v2/public/js/drag_and_drop.js:1528]: I try to keep consistency on names, on other part of the code has the same name also it responds as a JQuery type variable

File: drag_and_drop_v2/public/css/drag_and_drop.css
Comment (lcamacho) [reply drag_and_drop_v2/public/css/drag_and_drop.css]: fixed

**Comment by Agrendalath** [comment]
@sarina, we don't have any CCs for this XBlock yet. Good point, I'll prepare the nominations.

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (farhaanbukhsh) [inline drag_and_drop_v2/public/js/drag_and_drop.js:1628]:
```suggestion
                if (configuration.display_zone_borders_dragging && last_target_zone) {
                        last_target_zone.removeClass('zone-border');
                        last_target_zone = null;
                }
```

Wouldn't this be cleaner?

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (farhaanbukhsh) [inline drag_and_drop_v2/public/js/drag_and_drop.js:1541]:
```suggestion
                    } else if (last_target_zone) {
                            last_target_zone.removeClass('zone-border');
                            last_target_zone = null;
                        }
```

**Review by farhaanbukhsh — CHANGES_REQUESTED** [review]
Bunch of nits, we are almost there :)

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (lcamacho) [reply drag_and_drop_v2/public/js/drag_and_drop.js]: Fixed

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (lcamacho) [reply drag_and_drop_v2/public/js/drag_and_drop.js]: Fixed

**Review by farhaanbukhsh — APPROVED** [review]
👍 

- ✅ I tested this: on the devstack and the draggable border is appearing as well as there is no padding with the image :)
- ✅  I read through the code
- ❌  I checked for accessibility issues
- ❌  Includes documentation
- ❌  I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

**Comment by lcamacho** [comment]
@farhaanbukhsh one last review I added a commit to update version and changelog

**Comment by farhaanbukhsh** [comment]
@lcamacho  I have already reviewed it :) this can go for upstream review :)

**Comment by natabene** [comment]
@Agrendalath When you have a chance please review and merge.

**Comment by natabene** [comment]
@Agrendalath Just bumping this up in your queue.

**Comment by giovannicimolin** [comment]
@natabene I think the OSPR ticket got accidentally assigned to me. Is the assignment correct or should I reassign it?

**Comment by Agrendalath** [comment]
@natabene, thank you for the nudge. I've scheduled it for the beginning of the next week.

@giovannicimolin, I see that [OSPR-6468](https://openedx.atlassian.net/browse/OSPR-6468) is assigned to me. Have I missed something?

**Comment by farhaanbukhsh** [comment]
> @farhaanbukhsh one last review I added a commit to update version and changelog

This is good for upstream review 👍🏾

**Review by Agrendalath — CHANGES_REQUESTED** [review]

**Comment by edx-webhook** [comment]
@lcamacho Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/xblock-drag-and-drop-v2&entry.867055334=https://github.com/openedx/xblock-drag-and-drop-v2/pull/280&entry.1484655318=https://github.com/lcamacho&entry.752974735=2022-02-17+16:48&entry.1917517419=2022-03-25+20:06&entry.2133058324=No) so we can improve your experience in the future.

**Comment by edx-webhook** [comment]
@lcamacho Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/xblock-drag-and-drop-v2&entry.867055334=https://github.com/openedx/xblock-drag-and-drop-v2/pull/280&entry.1484655318=https://github.com/lcamacho&entry.752974735=2022-02-17+16:48&entry.1917517419=2022-03-25+20:06&entry.2133058324=No) so we can improve your experience in the future.

**Comment by natabene** [comment]
@lcamacho Apologies, I accidentally closed. Re-opened now.

File: drag_and_drop_v2/public/css/drag_and_drop.css
Comment (lcamacho) [reply drag_and_drop_v2/public/css/drag_and_drop.css:187]: This in combination with position absolute make the spinner appear at the center of the item

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (lcamacho) [reply drag_and_drop_v2/public/js/drag_and_drop.js:388]: Not sure how we can let the user know the show answer button is doing something if deleted

File: drag_and_drop_v2/public/css/drag_and_drop.css
Comment (lcamacho) [reply drag_and_drop_v2/public/css/drag_and_drop.css:865]: Done

File: drag_and_drop_v2/public/css/drag_and_drop.css
Comment (lcamacho) [reply drag_and_drop_v2/public/css/drag_and_drop.css:866]: Done

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (lcamacho) [reply drag_and_drop_v2/public/js/drag_and_drop.js:1528]: Done

File: drag_and_drop_v2/public/js/drag_and_drop_edit.js
Comment (lcamacho) [reply drag_and_drop_v2/public/js/drag_and_drop_edit.js:680]: Hawkeye done

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (lcamacho) [reply drag_and_drop_v2/public/js/drag_and_drop.js:820]: Yes, we need it since this class is a copy of what version 1 uses to wrap action buttons

File: Changelog.md
Comment (lcamacho) [reply Changelog.md:9]: Done

File: Changelog.md
Comment (lcamacho) [reply Changelog.md:11]: Done

File: Changelog.md
Comment (lcamacho) [reply Changelog.md:10]: done

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (Agrendalath) [reply drag_and_drop_v2/public/js/drag_and_drop.js:388]: @lcamacho, it would still have text, wouldn't it?

image

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @lcamacho! I've created [OSPR-6566](https://openedx.atlassian.net/browse/OSPR-6566) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (lcamacho) [reply drag_and_drop_v2/public/js/drag_and_drop.js]: Ok, now it says `Processing...` instead of showing the spinner

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @lcamacho! I've created [OSPR-6568](https://openedx.atlassian.net/browse/OSPR-6568) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: drag_and_drop_v2/public/js/drag_and_drop.js
Comment (Agrendalath) [reply drag_and_drop_v2/public/js/drag_and_drop.js:388]: @lcamacho, sorry, I've referenced an incorrect part of the code. I've meant the `fa-refresh` icon of the "Reset" button and the `fa-info-circle` icon of the "Show Answer" one. However, it would need some more reworks to look good natively, so I've reverted this change. We'll just change this through a custom theme.

**Comment by Agrendalath** [comment]
@Agrendalath, mentioning myself here to list this PR in GH API queries.

**Pull request merged**
